### ISSUE TITLE
Inconsistent reporting of kubernetes node memory-related metrics

### ISSUE THREAD

**Issue by walterdolce** [elastic/beats#8360]
I was trying to display the memory available vs memory usage of a Kubernetes node but then I noticed something strange. I am not sure whether it's metricsbeat that is at fault or not so I thought I would post it here.

<img width="1810" alt="screen shot 2018-09-19 at 17 47 26" src="https://user-images.githubusercontent.com/6195629/45768238-406ce600-bc34-11e8-9376-6e61ccd697e7.png">

As you can see the `kubernetes.node.memory.usage.bytes` is higher than the `kubernetes.node.memory.allocatable.bytes`. How is that possible?

The unit of measure configured within Grafana (`5.2.3`) is `bytes`. The version of metricbeat is `6.4.0`. The node is a Google Compute Engine instance of type `n1-standard-2 (2 vCPUs, 7.5 GB memory)`.

**Comment by exekias** [comment]
Hi @walterdolce, here you have an explanation of how allocatable memory works: https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable, as you can see, it's the memory available for pods, so it doesn't include the one reserved for kubernetes or used by other system parts.

If you sum the memory used by all the pods in the node it should be less or equal to the total allocatable. While the total memor used in the node includes these system & kuberentes related usages, which can end in what  you are experiencing.

**Comment by walterdolce** [comment]
Hi @exekias, thank you for your reply.

I see. Shouldn't `kubernetes.node.memory.allocatable.bytes` be called kubernetes.node.memory.allocatable_for_pods.bytes` instead, then? Or shouldn't the [documentation](https://www.elastic.co/guide/en/beats/metricbeat/current/exported-fields-kubernetes.html) say `Node allocatable memory for pods in bytes`, rather than `Node allocatable memory in bytes`?

I find the name of this metric (and its description) misleading now that I know its true meaning.

**Comment by exekias** [comment]
That's a fair point, I'm reopening this to improve the description for that field, more details here: https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable

**Comment by Baklap4** [comment]
Coming back on this issue with Metricbeat 8.2.2
We've plotted a graph with `kubernetes.node.memory.usage.bytes` and verified this against the actual usage `kubectl top nodes` reported.
Output of `kubectl top nodes` 5345 Mi. Whilst `kubernetes.node.memory.usage.bytes` reported 13GB.

We resorted to using `kubernetes.node.memory.workingset.bytes` which reported the actual used bytes: 5345 Mi
is this still an issue?
